docs/cli-reference/dfx-canister.mdx

```diff
@@ -280,6 +280,7 @@ You can use the following options with the `dfx canister create` command.
 | `--memory-allocation <memory>`            | Specifies how much memory the canister is allowed to use in total. This should be a value in the range [0..12 GiB]. A setting of 0 means the canister will have access to memory on a “best-effort” basis: It will only be charged for the memory it uses, but at any point in time may stop running if it tries to allocate more memory when there isn’t space available on the subnet. |
 | `--reserved-cycles-limit <limit>`         | Specifies the upper limit for the canister's reserved cycles.                                                                                                                                                                                                                                                                                                                            |
 | `--wasm-memory-limit <limit>`             | Specifies a soft upper limit for the canister's heap memory.                                                                                                                                                                                                                                                                                                                             |
+| `--log-viewer <principal>`                | Specifies the the principal of the log viewer of the canister. Can be specified more than once.                                                                                                                                                                                                                                                                                          |
 | `--log-visibility <visibility>`           | Specifies who is allowed to read the canister's logs. Can be either "controllers" or "public".                                                                                                                                                                                                                                                                                           |
```

Comment:
> Can be either "controllers" or "public".

This does not seem to be the case any more, since it can be `allowed_viewers`. And it's not clear that `--log-viewer` modifies `log_visibility`.

Reply:
I'd use `Specifies the principal as an allowed viewer. Can be specified more than once. Cannot be used with `--log-visibility`.  
As `a list of principals` might make users think they only need this parameter `once` to specify multiple principals. It's also consistent with the document for `--controller`.

Reply:
Fixed at https://github.com/dfinity/sdk/pull/3921/commits/201e2b727e6f000da193c446a0d68d4914eb3b0c.